tee-worker/ts-tests/stress/bin/stress.ts

```diff
@@ -0,0 +1,302 @@
+import {
+    ApiPromise as ParachainApiPromise,
+    WsProvider as ParachainWsProvider,
+    definitions as teeTypes,
+} from "parachain-api";
+import { Index } from "@polkadot/types/interfaces";
+import { z } from "zod";
+import { readFileSync } from "fs";
+import path, { dirname } from "path";
+import { fileURLToPath } from "url";
+import { ethers } from "ethers";
+import {
+    initWorkerConnection,
+    getSidechainMetadata,
+    getEnclave,
+    buildIdentityFromWallet,
+    getSidechainNonce,
+    Wallet,
+    Api,
+} from "../src/api";
+import { ContextManager } from "src/context-manager";
+import { processQueue, repeat } from "src/job-queue";
+import { Measurement, newMeasurementTracker } from "src/measurement";
+import { linkIdentity, requestVc1, requestVc4, setShieldingKey } from "src/steps";
+
+function logLine(...data: unknown[]): void {
+    process.stderr.write(`\n${data.join()}\n`);
+}
+
+function sleep(ms: number) {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+async function withRetry<T>(maxAttempts: number, task: () => Promise<T>): Promise<T> {
+    if (maxAttempts < 1) {
+        throw new Error(`Invalid number of attempts: ${maxAttempts}`);
+    }
+    let attemptsLeft = maxAttempts;
+    let waitMilliseconds = 1000;
+    let errors: unknown[] = [];
+    while (true) {
+        try {
+            return await task();
+        } catch (error) {
+            errors.push(error);
+        }
+        if (--attemptsLeft < 1) {
+            const messages = errors.map((error) =>
+                error instanceof Error ? error.message : JSON.stringify(error)
+            );
+            throw new Error(`Failed after ${maxAttempts} attempts:\n${messages.join("\n")}`);
+        }
+        waitMilliseconds *= 1 + Math.random(); // randomize backoff to avoid herd effect
+        await sleep(waitMilliseconds);
+    }
+}
+
+const Config = z.object({
+    timerEpoch: z.number(), // Arbitrary epoch to avoid precision loss when exporting/importing time info
+    connections: z.number().int().positive(),
+    iterations: z.number(),
+    substrateEndpoint: z.string().url(),
+    workerEndpoint: z.string().url(),
+    waitMilliseconds: z
+        .object({ min: z.number().nonnegative(), max: z.number().nonnegative() })
+        .refine(({ min, max }) => min <= max),
+});
+type Config = z.infer<typeof Config>;
+
+function loadConfig(path: string): Config {
+    return Config.parse(JSON.parse(readFileSync(path, { encoding: "utf8" })));
+}
+
+function randomSubstrateWallet(): KeyringPair {
+    const keyring = new Keyring({ type: "sr25519" });
+    return keyring.addFromSeed(randomBytes(32));
+}
+function randomEvmWallet(): ethers.Wallet {
+    return ethers.Wallet.createRandom();
+}
+function randomWallet(): Wallet {
+    return Math.random() > 0.5
+        ? { type: "substrate", keyringPair: randomSubstrateWallet() }
+        : { type: "evm", wallet: randomEvmWallet() };
+}
+
+const apiContextManager = (
```

Comment:
[`using`](https://github.com/tc39/proposal-explicit-resource-management) at home.

The test needs to establish several connections, create new ones on the fly in case of failure, and avoid leaking resources. These `ContextManager` objects help with that.